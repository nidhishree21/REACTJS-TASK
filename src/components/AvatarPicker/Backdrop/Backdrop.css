.Backdrop {
  position: fixed;
  z-index: 100;
  height: 100vmax;
  width: 100vmax;
  left: 0;
  top: 0;
  background-color: rgba(0, 0, 0, 0);
}
