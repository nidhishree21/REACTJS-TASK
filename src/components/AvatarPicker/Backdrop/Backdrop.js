import React from "react";
import PropTypes from "prop-types";
import styles from "./Backdrop.css";

const Backdrop = props =>
  props.show ? <div className={styles.Backdrop} onClick={props.hide} /> : null;

Backdrop.propTypes = {
  show: PropTypes.bool.isRequired,
  hide: PropTypes.func.isRequired
};
export default Backdrop;
