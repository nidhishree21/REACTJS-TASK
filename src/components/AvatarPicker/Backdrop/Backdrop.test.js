import React from "react";
import { shallow } from "enzyme";
import Backdrop from "./Backdrop";

const testProps = {
  show: true,
  hide: jest.fn(() => {})
};

describe("Backdrop", () => {
  it("renders Backdrop correctly", () => {
    let wrapper = shallow(<Backdrop {...testProps} />);
    expect(wrapper).toMatchSnapshot("1. Before - Show Backdrop");
    testProps.show = false;
    wrapper = shallow(<Backdrop {...testProps} />);
    expect(wrapper).toMatchSnapshot("2. After - Hide Backdrop");
  });
});
