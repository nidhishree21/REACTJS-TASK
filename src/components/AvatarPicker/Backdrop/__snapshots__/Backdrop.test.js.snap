// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. Before - Show Backdrop 1`] = `
<div
  className="Backdrop"
  onClick={[Function]}
/>
`;

exports[`2. After - Hide Backdrop 1`] = `""`;
